src/core/mac/sub_mac.cpp

```diff
@@ -565,6 +644,59 @@ void SubMac::SetState(State aState)
     }
 }
 
+void SubMac::SetMacKey(uint8_t aKeyIdMode, const uint8_t *aPrevKey, const uint8_t *aCurrKey, const uint8_t *aNextKey)
+{
+    switch (aKeyIdMode)
+    {
+    case Frame::kKeyIdMode0:
+    case Frame::kKeyIdMode2:
+        break;
+    case Frame::kKeyIdMode1:
+        OT_ASSERT(aPrevKey != NULL && aCurrKey != NULL && aNextKey != NULL);
+
+        memcpy(mPrevKey, aPrevKey, kMacKeySize);
+        memcpy(mCurrKey, aCurrKey, kMacKeySize);
+        memcpy(mNextKey, aNextKey, kMacKeySize);
+
+        break;
+
+    default:
+        OT_ASSERT(false);
+        break;
+    }
+
+    VerifyOrExit(!ShouldHandleTransmitSecurity(), OT_NOOP);
+
+    Get<Radio>().SetMacKey(aKeyIdMode, kMacKeySize, aPrevKey, aCurrKey, aNextKey);
+
+exit:
+    return;
+}
+
+void SubMac::SetMacKeyId(uint8_t aKeyIdMode, uint8_t aKeyId)
```

Comment:
I noticed in all these methods, we pass in `aKeyIdMode` but I think we practically only call it with `kKeyMode1` and only in this mode the code does something (saves the keys, etc). Do we think in future we may use other key id modes? Can we consider removing the `aKeyIdMode` parameter?

Reply:
I leave the key ID mode in the function parameter for future expansion/code reorg. I think if we keep key ID mode 0 and 2 handling in SubMac, too, we can remove the function `Mac::ProcessTransmitSecurity` from `Mac` layer. This will keep a cleaner code architecture so that the security related code does not scatter around `Mac` and `SubMac`. How do you think?

Reply:
SGTM.